gbasis/contractions.py

```diff
@@ -247,3 +250,36 @@ def coeffs(self, coeffs):
             raise ValueError("Coefficients array must have the same size as exponents array.")
 
         self._coeffs = coeffs
+
+
+def cartesian_gaussian_norm(components, exponent):
+    r"""Compute the normalization constant for a Cartesian Gaussian primitive.
+
+        .. math::
+
+            N(\vec{a}, \alpha) = (2 * \alpha / \pi)^{3/4} *
+            (4 * \alpha)^{(a_x + a_y + a_z)/2} /
+            ((2 * a_x - 1)!! * (2 * a_y - 1)!! * (2 * a_z - 1)!!)^{1/2}
+
+    Parameters
+    ----------
+    components : np.ndarray(3,)
+        The Cartesian components for angular momentum of the Gaussian primitive.
+        .. math::
+
+            \vec{a} = a_x + a_y + a_z
+
+    exponent: float
+        The exponent of the Cartesian Gaussian primitive.
+
+    Returns
+    -------
+    norm : float
+        The normalization constant of the Cartesian Gaussian primitive.
+
+    """
+    return (
+        ((2 * exponent / pi) ** (3 / 4))
+        * ((4 * exponent) ** (np.sum(components) / 2))
+        / (sqrt(np.prod(np.array([factorial2(2 * a - 1) for a in components]))))
```

Comment:
Try to support numpy operations. This would mean we want to make functions that can utilize arrays as well as floats.
In this case, `sqrt` should be replaced with `np.sqrt` and `factorial` applied directly to a numpy array (returning a numpy array) rather than using it within list comprehension (then converting it to numpy array).

Reply:
Fixed in latest commit.